File: frontend/types/workflows.ts
export interface RejectionAppeal {
  phones: string[];
  emails: string[];
  urls: string[];
}

export interface RejectionAnalysis {
  isRejection: boolean;
  reasonStated: boolean;
  reason: string | null;
  relatedDocuments: string[];
  suggestedActions: string[];
  appeal: RejectionAppeal;
  confidence: number;
}

export interface VerificationSignal {
  type: string;
  detected: boolean;
  value: string | null;
}

export interface VerificationResult {
  signals: VerificationSignal[];
}

export interface FormFieldSuggestion {
  key: string;
  sensitive: boolean;
  suggestedValue: string | null;
  source: string | null;
  confidence: number;
}

export interface FormAnalysis {
  isForm: boolean;
  fields: FormFieldSuggestion[];
}

File: backend/app/schemas/workflows.py
from pydantic import BaseModel


class RejectionAppeal(BaseModel):
    phones: list[str] = []
    emails: list[str] = []
    urls: list[str] = []


class RejectionRead(BaseModel):
    isRejection: bool
    reasonStated: bool
    reason: str | None = None
    relatedDocuments: list[str] = []
    suggestedActions: list[str] = []
    appeal: RejectionAppeal
    confidence: float


class VerificationSignal(BaseModel):
    type: str
    detected: bool
    value: str | None = None


class VerificationRead(BaseModel):
    signals: list[VerificationSignal] = []


class FormField(BaseModel):
    key: str
    sensitive: bool = False
    suggestedValue: str | None = None
    source: str | None = None
    confidence: float


class FormRead(BaseModel):
    isForm: bool
    fields: list[FormField] = []
